packages/drivers/driver-web-cache/src/FluidCache.ts

```diff
@@ -117,10 +120,35 @@ export class FluidCache implements IPersistedCache {
 		});
 	}
 
+	private async openDb() {
+		if (this.db === undefined) {
+			this.db = await getFluidCacheIndexedDbInstance(this.logger);
+			// Need to close the db on version change if opened.
+			this.db.addEventListener("versionchange", (ev) => {
+				this.db?.close();
+				this.db = undefined;
+			});
+			this.db.addEventListener("close", (ev) => {
+				clearTimeout(this.dbCloseTimer);
+				this.dbCloseTimer = undefined;
+				this.db = undefined;
+			});
+			// Schedule db close after 30s.
+			if (this.dbCloseTimer === undefined) {
```

Comment:
other than the race condition I mentioned above, sounds like this.dbCloseTimer should be always undefined here, no?
If that's true, maybe assert is better? Not sure.